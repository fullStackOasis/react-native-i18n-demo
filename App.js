/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import I18n from './i18n/i18n';

import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import LocalizedHeader from './LocalizedHeader';
import LocalizedReloadInstructions from './LocalizedReloadInstructions';
import {
  LearnMoreLinks,
  Colors,
  DebugInstructions
} from 'react-native/Libraries/NewAppScreen';

const App: () => React$Node = () => {
	/* Debug?
	import { getLanguages } from 'react-native-i18n';

	getLanguages().then(languages => {
		console.log("showing languages");
		console.log(languages); // ['en-US', 'en']
	});
	*/
	//I18n.defaultLocale = "fr";
	// For test purposes only, set the locale, and see your text change to the new language.
	// I18n.locale = 'fr';
	return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={styles.scrollView}>
          <LocalizedHeader text={I18n.t('welcome')}/>
          {global.HermesInternal == null ? null : (
            <View style={styles.engine}>
              <Text style={styles.footer}>Engine: Hermes</Text>
            </View>
          )}
          <View style={styles.body}>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>{I18n.t('step_one')}</Text>
              <Text style={styles.sectionDescription}>
			  {I18n.t('edit_instr1')} <Text style={styles.highlight}>{I18n.t('edit_instr2')}</Text> {I18n.t('edit_instr3')}
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>{I18n.t('see_results')}</Text>
              <Text style={styles.sectionDescription}>
                <LocalizedReloadInstructions />
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>{I18n.t('debug')}</Text>
              <Text style={styles.sectionDescription}>
                <DebugInstructions />
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>{I18n.t('learn_more')}</Text>
              <Text style={styles.sectionDescription}>
			  {I18n.t('do_next')}
              </Text>
            </View>
            <LearnMoreLinks />
          </View>
        </ScrollView>
      </SafeAreaView>
    </>
  );
};

const styles = StyleSheet.create({
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  engine: {
    position: 'absolute',
    right: 0,
  },
  body: {
    backgroundColor: Colors.white,
  },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  highlight: {
    fontWeight: '700',
  },
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  },
});

export default App;
