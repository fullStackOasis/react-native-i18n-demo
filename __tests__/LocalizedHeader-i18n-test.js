/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import LocalizedHeader from '../LocalizedHeader';
import I18n from '../i18n/i18n';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('LocalizedHeader renders strings in **French**', () => {
	I18n.locale = 'fr';
	const h = renderer
		.create(<LocalizedHeader text={I18n.t('welcome')}></LocalizedHeader>)
		.toJSON();
	expect(h).toMatchSnapshot();
});

it('LocalizedHeader renders strings in **English**', () => {
	I18n.locale = 'en';
	const h = renderer
		.create(<LocalizedHeader text={I18n.t('welcome')}></LocalizedHeader>)
		.toJSON();
	expect(h).toMatchSnapshot();
});
  