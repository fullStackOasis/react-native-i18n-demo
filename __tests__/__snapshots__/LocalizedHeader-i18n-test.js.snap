// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedHeader renders strings in **English** 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "paddingBottom": 40,
      "paddingHorizontal": 32,
      "paddingTop": 96,
    }
  }
>
  <Image
    accessibilityRole="image"
    source={
      Object {
        "testUri": "../Libraries/NewAppScreen/components/logo.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "marginBottom": -192,
          "marginLeft": -128,
          "opacity": 0.2,
          "overflow": "visible",
          "resizeMode": "cover",
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 40,
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    Welcome to React
  </Text>
</View>
`;

exports[`LocalizedHeader renders strings in **French** 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "paddingBottom": 40,
      "paddingHorizontal": 32,
      "paddingTop": 96,
    }
  }
>
  <Image
    accessibilityRole="image"
    source={
      Object {
        "testUri": "../Libraries/NewAppScreen/components/logo.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "marginBottom": -192,
          "marginLeft": -128,
          "opacity": 0.2,
          "overflow": "visible",
          "resizeMode": "cover",
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 40,
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    Bienvenue chez React
  </Text>
</View>
`;
